src/hints/dict/initSquashData.ts

```diff
@@ -0,0 +1,97 @@
+import { z } from 'zod';
+
+import { ExpectedFelt } from 'errors/primitives';
+
+import { Felt } from 'primitives/felt';
+import { isFelt } from 'primitives/segmentValue';
+import { VirtualMachine } from 'vm/virtualMachine';
+
+import { HintName } from 'hints/hintName';
+import {
+  CellRef,
+  cellRef,
+  resOperand,
+  ResOperand,
+} from 'hints/hintParamsSchema';
+
+/** Zod object to parse InitSquashData hint */
+export const initSquashDataParser = z
+  .object({
+    InitSquashData: z.object({
+      dict_accesses: resOperand,
+      ptr_diff: resOperand,
+      n_accesses: resOperand,
+      big_keys: cellRef,
+      first_key: cellRef,
+    }),
+  })
+  .transform(
+    ({
+      InitSquashData: {
+        dict_accesses,
+        ptr_diff,
+        n_accesses,
+        big_keys,
+        first_key,
+      },
+    }) => ({
+      type: HintName.InitSquashData,
+      dictAccesses: dict_accesses,
+      ptrDiff: ptr_diff,
+      nAccesses: n_accesses,
+      bigKeys: big_keys,
+      firstKey: first_key,
+    })
+  );
+
+/**
+ * InitSquashData hint
+ *
+ * Initialize the squashed dictionnary data.
+ */
+export type InitSquashData = z.infer<typeof initSquashDataParser>;
+
+/**
+ * Initialize the squashed dictionnary
+ *
+ * - Insert all accessed keys
+ * - Assert the biggest key to `bigKeys` address
+ * - Assert the smallest (first) key to `firstKey` address
+ *
+ */
```

Comment:
JSDoc should mainly provide information on the expected parameters. As a first read I don't know what `bigKeys` is